video/out/opengl/context_glx.c

```diff
@@ -134,17 +99,50 @@ static bool create_context_x11_gl3(struct ra_ctx *ctx, GL *gl, int gl_version,
     }
 
     int context_attribs[] = {
-        GLX_CONTEXT_MAJOR_VERSION_ARB, MPGL_VER_GET_MAJOR(gl_version),
-        GLX_CONTEXT_MINOR_VERSION_ARB, MPGL_VER_GET_MINOR(gl_version),
+        GLX_CONTEXT_MAJOR_VERSION_ARB, 0,
+        GLX_CONTEXT_MINOR_VERSION_ARB, 0,
         GLX_CONTEXT_PROFILE_MASK_ARB, profile_mask,
         GLX_CONTEXT_FLAGS_ARB, ctx_flags,
         None
     };
-    vo_x11_silence_xlib(1);
-    GLXContext context = glXCreateContextAttribsARB(vo->x11->display,
-                                                    p->fbc, 0, True,
-                                                    context_attribs);
-    vo_x11_silence_xlib(-1);
+
+    GLXContext context;
+
+    if (!es) {
+	for (int n = 0; mpgl_preferred_gl_versions[n]; n++) {
+            int version = mpgl_preferred_gl_versions[n];
+            MP_VERBOSE(ctx, "Creating OpenGL %d.%d context...\n",
+                       MPGL_VER_P(version));
+            if (version >= 320) {
+	        if (!has_new_ctx)
+                    continue;
+
+                context_attribs[1] = MPGL_VER_GET_MAJOR(version);
+                context_attribs[3] = MPGL_VER_GET_MINOR(version);
+
+                vo_x11_silence_xlib(1);
+                context = glXCreateContextAttribsARB(vo->x11->display,
+                                                     p->fbc, 0, True,
+                                                     context_attribs);
+                vo_x11_silence_xlib(-1);
+	    } else {
+                context = glXCreateNewContext(vo->x11->display, p->fbc,
+                                              GLX_RGBA_TYPE, NULL, True);
+	    }
+
+            if (context)
+                break;
+	}
```

Comment:
Looks like some tabs/spacing went wonky in this section.

Reply:
Nicely spotted will fix. Is it worth adding some [.editorconfig](https://editorconfig.org/) files in-tree so that formatting is correctly picked and applied by the editor?

Reply:
I personally don't think so, but that can be another PR or something is someone thinks it is worth it.

Reply:
Personally I'd be fine with editorconfig files. Usually there's only one/few of them, and it can only help. We don't use modlines anyway, and editorconfig is generally with good support in many editors. But in a different PR.

Reply:
Fixed the tabs issue. Will kick off the editorconfig bikeshed in separate PR at later point